sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3804,6 +3804,13 @@ class Dataset[T] private[sql](
     } catch {
       case _: ParseException => throw QueryCompilationErrors.invalidViewNameError(viewName)
     }
+
+    if (tableIdentifier.database.isDefined) {
+      throw new AnalysisException(
+        errorClass = "TEMP_VIEW_DOES_NOT_BELONG_TO_A_DATABASE",
```

Comment:
In the parser side, we have this
```
      val tableIdentifier = visitMultipartIdentifier(ctx.multipartIdentifier).asTableIdentifier
      if (tableIdentifier.database.isDefined) {
        // Temporary view names should NOT contain database prefix like "database.table"
        throw QueryParsingErrors
          .notAllowedToAddDBPrefixForTempViewError(tableIdentifier.database.get, ctx)
      }
```
shall we reuse the error class here?

Reply:
And this is a breaking change, can we add a legacy config to restore the previous behavior? (ignore the database)

Reply:
What should be default behavior? By default ignore the database or throw exception?

Reply:
I think throwing exception is a better default.

Reply:
The config idea sounds good.

I found the parser side is using `_LEGACY_ERROR_TEMP_0054`.

```
  def notAllowedToAddDBPrefixForTempViewError(
      database: String,
      ctx: CreateViewContext): Throwable = {
    new ParseException(
      errorClass = "_LEGACY_ERROR_TEMP_0054",
      messageParameters = Map("database" -> database),
      ctx)
  }
```

We can use the same error message body but at least we should rename this error class name.

Reply:
Ok I basically
1. add a config to control the behavior. By default we throw exception.
2. reuse `_LEGACY_ERROR_TEMP_0054` but rename to a meaningful name.